lib/src/plugin/HttpInteractions.dart

```diff
@@ -0,0 +1,24 @@
+import 'package:nyxx/nyxx.dart';
+
+import 'package:nyxx/src/utils/iterable_extension.dart';
+
+abstract class HttpInteractionsPlugin implements NyxxPlugin<NyxxRest> {
```

Comment:
added Plugin to minimize possibility of confusion between our abstract and actual implementation